lib/src/architectureIntegrationTest/kotlin/com/lemonappdev/konsist/circulardependency/CircularDependencyTest.kt

```diff
@@ -0,0 +1,153 @@
+package com.lemonappdev.konsist.circulardependency
+
+import com.lemonappdev.konsist.api.Konsist
+import com.lemonappdev.konsist.core.architecture.Layer
+import com.lemonappdev.konsist.core.exception.KoPreconditionFailedException
+import org.amshove.kluent.shouldThrow
+import org.amshove.kluent.withMessage
+import org.junit.jupiter.api.Test
+
+class CircularDependencyTest {
+    @Test
+    fun `simply circular dependency with two layers`() {
+        // given
+        val layer1 = Layer("layer1", "layer1")
+        val layer2 = Layer("layer2", "layer2")
+
+        // when
+        val sut = {
+            Konsist
+                .architecture(layer1, layer2)
+                .addDependencies {
+                    layer1.dependsOn(layer2)
+                    layer2.dependsOn(layer1)
+                }
+        }
+
+        // then
+        sut shouldThrow KoPreconditionFailedException::class withMessage """
+            Illegal circular dependencies:
+            Layer(name=layer2, isDefinedBy=layer2) -->
+            Layer(name=layer1, isDefinedBy=layer1) -->
+            Layer(name=layer2, isDefinedBy=layer2).
+        """.trimIndent()
+    }
+
+    @Test
+    fun `simply circular dependency with three layers`() {
+        // given
+        val layer1 = Layer("layer1", "layer1")
+        val layer2 = Layer("layer2", "layer2")
+        val layer3 = Layer("layer3", "layer3")
+
+        // when
+        val sut = {
+            Konsist
+                .architecture(layer1, layer2, layer3)
+                .addDependencies {
+                    layer1.dependsOn(layer2)
+                    layer2.dependsOn(layer3)
+                    layer3.dependsOn(layer1)
+                }
+        }
+
+        // then
+        sut shouldThrow KoPreconditionFailedException::class withMessage """
+            Illegal circular dependencies:
+            Layer(name=layer3, isDefinedBy=layer3) -->
+            Layer(name=layer1, isDefinedBy=layer1) -->
+            Layer(name=layer2, isDefinedBy=layer2) -->
+            Layer(name=layer3, isDefinedBy=layer3).
+        """.trimIndent()
+    }
+
+    @Test
+    fun `complex circular dependency 1`() {
```

Comment:
1. Let's name these tests "circular dependency 1...5" 
2. Let's move them to separate packages like architecture1..5